File: software/examples/php/ExampleThreshold.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletDistanceIR.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletDistanceIR;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Distance IR Bricklet

// Callback function for distance reached callback (parameter has unit mm)
function cb_distanceReached($distance)
{
    echo "Distance: " . $distance/10.0 . " cm\n";
}

$ipcon = new IPConnection(); // Create IP connection
$dir = new BrickletDistanceIR(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
$dir->setDebouncePeriod(10000);

// Register distance reached callback to function cb_distanceReached
$dir->registerCallback(BrickletDistanceIR::CALLBACK_DISTANCE_REACHED,
                       'cb_distanceReached');

// Configure threshold for distance "smaller than 30 cm" (unit is mm)
$dir->setDistanceCallbackThreshold('<', 30*10, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/python/example_threshold.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_distance_ir import DistanceIR

# Callback function for distance smaller than 30 cm (parameter has unit mm)
def cb_distance_reached(distance):
    print('Distance: ' + str(distance/10.0) + ' cm')

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    dir = DistanceIR(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get threshold callbacks with a debounce time of 10 seconds (10000ms)
    dir.set_debounce_period(10000)

    # Register threshold reached callback to function cb_distance_reached
    dir.register_callback(dir.CALLBACK_DISTANCE_REACHED, cb_distance_reached)

    # Configure threshold for "smaller than 30 cm" (unit is mm)
    dir.set_distance_callback_threshold('<', 30*10, 0)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()
